src/StatisticalAnalyser.java

```diff
@@ -65,53 +65,64 @@ public double averageLinkWeight(ArrayList<ArrayList<Path>> paths) {
     }
 
     public double standardDeviationPathWeight(ArrayList<ArrayList<Path>> paths) {
+        LinkWeightTracker lwTracker = linkWeightTracker(paths);
         int nPaths = 0;
         double accumulatedDeviation = 0.0;
         double average = averagePathWeight(paths);
         for(ArrayList<Path> alp : paths) {
             nPaths += alp.size();
             for(Path path: alp) {
-                accumulatedDeviation += Math.pow((path.getWeight() - average),2);
+                accumulatedDeviation += Math.pow((lwTracker.weight(path) - average),2);
             }
         }
         return Math.sqrt(accumulatedDeviation/(double)nPaths);
     }
 
     public double averagePathWeight(ArrayList<ArrayList<Path>> paths) {
+        LinkWeightTracker lwTracker = linkWeightTracker(paths);
         double accumulatedPathWeight = 0.0;
         int nPaths = 0;
         for(ArrayList<Path> alp : paths) {
             nPaths += alp.size();
             for(Path path: alp)
-                accumulatedPathWeight += path.getWeight();
+                accumulatedPathWeight += lwTracker.weight(path);
         }
         return accumulatedPathWeight/(double)nPaths;
     }
 
     public double standardDeviationPathNormWeight(ArrayList<ArrayList<Path>> paths) {
+        LinkWeightTracker lwTracker = linkWeightTracker(paths);
         int nPaths = 0;
         double accumulatedDeviation = 0.0;
         double average = averagePathNormWeight(paths);
         for(ArrayList<Path> alp : paths) {
             nPaths += alp.size();
             for(Path path: alp) {
-                accumulatedDeviation += Math.pow((average - path.getWeight()/(double)(path.size()-1)),2);
+                accumulatedDeviation += Math.pow((average - lwTracker.weight(path)/(double)(path.size()-1)),2);
             }
         }
         return Math.sqrt(accumulatedDeviation/(double)nPaths);
     }
 
     public double averagePathNormWeight(ArrayList<ArrayList<Path>> paths) {
+        LinkWeightTracker lwTracker = linkWeightTracker(paths);
         double accumulatedPathWeight = 0;
         int nPaths = 0;
         for(ArrayList<Path> alp : paths) {
             nPaths += alp.size();
             for(Path path: alp)
-                accumulatedPathWeight += path.getWeight()/(double)(path.size()-1);
+                accumulatedPathWeight += lwTracker.weight(path)/(double)(path.size()-1);
         }
         return accumulatedPathWeight/(double)nPaths;
     }
 
+    private LinkWeightTracker linkWeightTracker(ArrayList<ArrayList<Path>> paths) {
+        LinkWeightTracker lwTracker = new LinkWeightTracker(graph);
+        for (ArrayList<Path> samePairPaths : paths) {
```

Comment:
The constructor of LinkWeightTracker need to receive the communication volume. Also, the constructor of StatisticalAnalyser also needs this change.